// Copyright 2017, Symphony Secure Communications.
// All rights reserved.

syntax = "proto3";

option objc_class_prefix = "PODDSC";

package pod_discovery;

// Interface exported by the server.
service PodDiscovery {

  // Obtain extensive details of a Pod based on Vanity name.
  rpc GetPod(PodName) returns (Pod) {}

  // A server-to-client streaming RPC.
  rpc ListPods(ListPodRequest) returns (stream Pod) {}

}

message PodName {
  string vanity_url = 1;
}

message ListPodRequest {
  string environment = 1;
}

message Pod {
  string pod_name = 1;
  string vanity_url = 2;
  string cluster = 3;
  string group = 4;
}
